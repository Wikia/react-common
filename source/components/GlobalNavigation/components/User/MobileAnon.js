import React from 'react';
import PropTypes from 'prop-types';

import Button from '../../../Button';
import Avatar from '../../../Avatar';

import withRedirectUrl from '../../hocs/withRedirectUrl';

const MobileAnon = ({ data, getUrlWithRedirect }) => (
    <Button
        className="wds-global-navigation__modal-control wds-global-navigation__modal-control-anon"
        href={getUrlWithRedirect(data.signin.href)}
        text
    >
        <Avatar />
    </Button>
);

MobileAnon.propTypes = {
    data: PropTypes.object.isRequired,
    getUrlWithRedirect: PropTypes.func.isRequired,
};

export default withRedirectUrl(MobileAnon);
