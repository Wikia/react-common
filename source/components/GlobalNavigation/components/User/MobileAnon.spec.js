import React from 'react';
import { shallow } from 'enzyme';
import merge from 'lodash/merge';

import MobileAnon from './MobileAnon';

const defaultProps = {
    model: {
        signin: {
            type: 'link-signin',
            href: 'https://www.wikia.com/signin',
            title: {
                type: 'translatable-text',
                key: 'global-navigation-user-sign-in',
            },
            'param-name': 'redirectto',
            'tracking-label': 'account.sign-in',
        },
    },
};

jest.mock('../../hocs/withRedirectUrl', () => Component => props => (
    <Component {...props} getUrlWithRedirect={(href, param) => `${href} + ${param}`} />
));

function renderComponent(props) {
    const computedProps = merge({}, defaultProps, props);

    return shallow(<MobileAnon {...computedProps} />);
}

test('MobileAnon renders correctly', () => {
    expect(renderComponent().dive()).toMatchSnapshot();
});
