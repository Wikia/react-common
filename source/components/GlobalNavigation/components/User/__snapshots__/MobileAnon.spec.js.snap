// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileAnon renders correctly 1`] = `
<Button
  className="wds-global-navigation__modal-control wds-global-navigation__modal-control-anon"
  disabled={false}
  fullwidth={false}
  href="https://www.wikia.com/signin + redirectto"
  light={false}
  onClick={[Function]}
  secondary={false}
  square={false}
  text={true}
>
  <Avatar
    alt="User avatar"
  />
</Button>
`;
